ethereum/core/src/main/java/tech/pegasys/teku/core/operationsignatureverifiers/VoluntaryExitSignatureVerifier.java

```diff
@@ -34,11 +33,7 @@ public boolean verifySignature(
     final VoluntaryExit exit = signedExit.getMessage();
 
     BLSPublicKey publicKey =
-        BeaconStateCache.getTransitionCaches(state)
-            .getValidatorsPubKeys()
-            .get(
-                exit.getValidator_index(),
-                idx -> state.getValidators().get(toIntExact(idx.longValue())).getPubkey());
+        ValidatorsUtil.getValidatorPubKey(state, exit.getValidator_index()).orElseThrow();
```

Comment:
Returning `false` here would be good too.

Reply:
The similar issue as with ProposerSlashingSignatureVerifier`. Also added to the issue #3319